### DISCUSSION TITLE
Is there a way within Typescript to have certain form props 'opt out' of recursive FieldPath detection?

### DISCUSSION THREAD

**Discussion by alexkatz** [react-hook-form/react-hook-form#8304] [Q&A] [answered]
For example, let's say I have form data that looks like this:

```typescript
type GiantNestedThing = {
  i: {
    dont: {
      care: {
        about: {
          these: {
            fields: {
              atAll: number;
            },
          },
        },
      },
    }, 
  },
};

type FormData = {
  firstName?: string;
  lastName?: string;
  thing?: GiantNestedThing;
}
```

Is there a way within Typescript to prevent keys from within `GiantNestedThing` to be gathered via `FieldPath` and suggested within `register`?

For example, I'd prefer not to see Typescript suggestions like
```typescript 
register('thing.i.dont.care.about')
register('thing.i.dont.care.about.these')
register('thing.i.dont.care.about.these.fields')
...
``` 
but rather have it simply suggest 
```typescript
register('firstName')
register('lastName')
register('thing')
``` 
and stop there.

Perhaps something like:

```typescript
type FormData = {
  firstName?: string;
  lastName?: string;
  thing?: IgnoreNestedFields<GiantNestedThing>;
}
```

Maybe there's already a way to do this that I'm missing, though I'm aware I can likely write my own tweaked implementation of the `Path` type and add an extra `IgnoreNestedFields` conditional within it, or something like that.

**Comment by bluebill1049** [comment, marked as answer]
I believe this is something that we are improving on the next major version by lazy evaluating the path. https://github.com/react-hook-form/react-hook-form/discussions/7433
